server.js

```diff
@@ -7,23 +7,60 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  }, 
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('A little home for my personal Happy Thoughts API')
+})
+
+// Get the most recent 20 thoughts
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
+
+// Post a new thought
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought(req.body).save()
+    res.status(200).json(newThought)
+  } catch (err) {
+    res.status(400).json({message:'Could not save thought', errors: err.errors})
+  }
+})
+
+// Add a heart to a thought
+app.put('/thoughts/:id/like', async (req,res) => {
+  try {
+    const thought = await Thought.findById(req.params.id)
+    await Thought.updateOne({_id: thought._id}, {hearts: thought.hearts + 1})
+    res.status(200).json(thought)
```

Comment:
One thing I noticed when trying this endpoint in Postman: 
When I try to like a thought with, say, 0 likes I can do that (which is good! 😄 ) But the JSON-response I'm getting back still says hearts: 0, even though when I do a GET request to /thoughts I can see that my like has reached the database. That's because in line 58 you send back the original thought from line 56 and not the updated one.

It would be nice to get the updated thought as a response and I think you could do it like this: 
```suggestion
    const thought = await Thought.findById(req.params.id)
    const updatedThought = await Thought.updateOne({_id: thought._id}, {hearts: thought.hearts + 1})
    res.status(200).json(updatedThought)
```

Reply:
Eagle Eye @HenrikeW 🥇